src/main.rs

```diff
@@ -4,19 +4,69 @@
 use core::marker::Sized;
 use esp_alloc as _;
 use esp_backtrace as _;
-use esp_println::println;
+use esp_hal::{gpio::AnyPin, interrupt::{software::SoftwareInterruptControl, Priority}, peripherals::UART1, rng::Rng, timer::timg::TimerGroup, uart::{Config, RxConfig, Uart}};
+use esp_hal_embassy::InterruptExecutor;
 
 use embassy_executor::Spawner;
-use ssh_stamp::serve::start;
+use ssh_stamp::espressif::{net::{accept_requests, if_up}, rng, buffered_uart::BufferedUart};
+use static_cell::StaticCell;
 
 #[esp_hal_embassy::main]
 async fn main(spawner: Spawner) -> ! {
     esp_alloc::heap_allocator!(size: 72 * 1024);
     esp_println::logger::init_logger_from_env();
 
-    let res = start(spawner).await;
-    if let Err(e) = res {
-        println!("Giving up: {:?}", e);
+    // System init
+    let peripherals = esp_hal::init({
+        esp_hal::Config::default()
+    });
+    let mut rng = Rng::new(peripherals.RNG);
+    let timg0 = TimerGroup::new(peripherals.TIMG0);
+
+    rng::register_custom_rng(rng);
+
+    cfg_if::cfg_if! {
+       if #[cfg(feature = "esp32")] {
+            let timg1 = TimerGroup::new(peripherals.TIMG1);
+            esp_hal_embassy::init(timg1.timer0);
+       } else {
+           use esp_hal::timer::systimer::SystemTimer;
+           let systimer = SystemTimer::new(peripherals.SYSTIMER);
+           esp_hal_embassy::init(systimer.alarm0);
+       }
     }
-    todo!(); // try again somehow
+
+    let wifi_controller = esp_wifi::init(timg0.timer0, rng, peripherals.RADIO_CLK).unwrap();
+
+    // Bring up the network interface and start accepting SSH connections.
+    let tcp_stack = if_up(spawner, wifi_controller, peripherals.WIFI, &mut rng).await.unwrap();
+
+    // Set up software buffered UART to run in a higher priority InterruptExecutor
+    let uart_buf = UART_BUF.init_with(BufferedUart::new);
+    let software_interrupts = SoftwareInterruptControl::new(peripherals.SW_INTERRUPT);
+    let interrupt_exeuctor = INT_EXECUTOR.init_with(|| InterruptExecutor::new(software_interrupts.software_interrupt0));
+    let interrupt_spawner = interrupt_exeuctor.start(Priority::Priority10);
```

Comment:
`Priority10` was arbitrarily chosen. I wasn't sure what other interrupt priorities are live in the system or how long they may block for, but this was high enough to work. :shrug:.

Reply:
LGTM, but I'm blind too about the dynamics of the system, so for now I'll just mention this in issue #28 and get back to it once we need to focus on squeezing perf out of this thing :)